test-network/scripts/addAnchorPeer.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# import utils
+. scripts/envVar.sh
+. scripts/configUpdate.sh
+
+
+# NOTE: this must be run in a CLI container since it requires jq and configtxlator 
+createAnchorPeerUpdate() {
+  ORG=$1
+  setGlobals $ORG
+
+  infoln "Fetching channel config for channel $CHANNEL_NAME"
+  fetchChannelConfig $ORG $CHANNEL_NAME ${CORE_PEER_LOCALMSPID}config.json
+
+  infoln "Generating anchor peer update transaction for Org${ORG} on channel $CHANNEL_NAME"
+
+  if [ $ORG -eq 1 ]; then
+    HOST="peer0.org1.example.com"
+    PORT=7051
+  elif [ $ORG -eq 2 ]; then
+    HOST="peer0.org2.example.com"
+    PORT=9051
+  elif [ $ORG -eq 3 ]; then
+    HOST="peer0.org3.example.com"
+    PORT=11051
+  else
+    errorln "Org${ORG} unknown"
+  fi
+
+  set -x
+  # Modify the configuration to append the anchor peer 
+  jq '.channel_group.groups.Application.groups.'${CORE_PEER_LOCALMSPID}'.values += {"AnchorPeers":{"mod_policy": "Admins","value":{"anchor_peers": [{"host": "'$HOST'","port": '$PORT'}]},"version": "0"}}' ${CORE_PEER_LOCALMSPID}config.json > ${CORE_PEER_LOCALMSPID}modified_config.json
+  { set +x; } 2>/dev/null
+
+  # Compute a config update, based on the differences between 
+  # {orgmsp}config.json and {orgmsp}modified_config.json, write
+  # it as a transaction to {orgmsp}anchors.tx
+  createConfigUpdate ${CHANNEL_NAME} ${CORE_PEER_LOCALMSPID}config.json ${CORE_PEER_LOCALMSPID}modified_config.json ${CORE_PEER_LOCALMSPID}anchors.tx
+}
+
+updateAnchorPeer() {
+  ORG=$1
+  CHANNEL_NAME=$2
+  setGlobalsCLI $ORG
+
+  peer channel update -o orderer.example.com:7050 --ordererTLSHostnameOverride orderer.example.com -c $CHANNEL_NAME -f ${CORE_PEER_LOCALMSPID}anchors.tx --tls --cafile $ORDERER_CA >&log.txt
+  res=$?
+  cat log.txt
+  verifyResult $res "Anchor peer update failed"
+  successln "Anchor peers updated for org '$CORE_PEER_LOCALMSPID' on channel '$CHANNEL_NAME'"
+}
+
+ORG=$1
```

Comment:
May be easier to consolidate these four lines into 

```
createAnchorPeerUpdate $1 $2

updateAnchorPeer $1 $2
```

rather than re-declare these variables so many times but your taste may vary.

Reply:
I don't have a strong preference here. I coded this both ways at various times but couldn't decide which I preferred or if one would be considered "best practice."

Reply:
Updated.

Reply:
Actually, I changed this up a bit. Consolidated down to setting the ORG and CHANNEL_NAME from $1 and $2 here, set the globals once for the script, and then avoid duplicating that code in each of the functions.